$OpenBSD: patch-src_decoder_plugins_FfmpegDecoderPlugin_cxx,v 1.4 2016/11/24 08:27:28 dcoppa Exp $

commit 77a9940461f96a608b1f564f810b146980e63c4d
Author: Max Kellermann <max@musicpd.org>
Date:   Wed Oct 26 18:06:56 2016 +0200

decoder/ffmpeg: ignore empty packets

An empty packet would be a command for avcodec_send_packet() to
finalize the codec.

Fixes https://bugs.musicpd.org/view.php?id=4588

--- src/decoder/plugins/FfmpegDecoderPlugin.cxx.orig	Wed Nov 23 16:04:04 2016
+++ src/decoder/plugins/FfmpegDecoderPlugin.cxx	Wed Nov 23 16:04:50 2016
@@ -643,7 +643,7 @@ FfmpegDecode(Decoder &decoder, InputStream &input,
 			/* end of file */
 			break;
 
-		if (packet.stream_index == audio_stream) {
+		if (packet.size > 0 && packet.stream_index == audio_stream) {
 			cmd = ffmpeg_send_packet(decoder, input,
 						 packet,
 						 *codec_context,
